### PR TITLE
Allow overriding PUSH_URL and RATE_LIMIT_URL in local.properties

### PR DESCRIPTION
**Pull request by pdecat**
This PR takes on @JBassett's [suggestion](https://github.com/home-assistant/android/pull/1010#pullrequestreview-502453327) to extract PUSH_URL in a properties file.

The same is done for RATE_LIMIT_URL.

### CODE REVIEW

**Comment by runningcode** [comment]
Why not use a standard gradle property stores in the gradle.properties file? Users can then override this by setting the same property in their home directory?
https://docs.gradle.org/current/userguide/build_environment.html#sec:gradle_configuration_properties

**Comment by pdecat** [comment]
Hi @runningcode, that's indeed an option but I'd rather have this project specific option in the project root and `gradle.properties` is already versioned there. So I went with `local.properties` which is already excluded in the `.gitignore` file.

I'm open to reconsider that decision if using `local.properties` is a bad option.

**Comment by runningcode** [comment]
The `local.properties` file is auto-generated by Android Studio.
You can store the default values int he project's root `gradle.properties` and then user's can override them with their gradle home's `~/.gradle/gradle.properties` file

**Comment by pdecat** [comment]
> The `local.properties` file is auto-generated by Android Studio which means it may be destroyed at any moment.

Ok, good point not to use that file then! I've pushed the requested change, please tell me if that's what you had in mind.

File: README.md [hunk 1]
```diff
@@ -43,7 +43,15 @@ If you get stuck while setting up your own environment, you can ask questions in
 
 ### Push Notifications
 
-If you want to work on push notifications or use a development build with push notifications please go the server side code [HERE](https://github.com/home-assistant/mobile-apps-fcm-push) and deploy it to your firebase project.  Once you have your androidV1 URL to the deployed service, exchange that for your local builds [PUSH_URL](https://github.com/home-assistant/android/blob/master/common/src/main/java/io/homeassistant/companion/android/common/data/integration/impl/IntegrationRepositoryImpl.kt#L43).
+If you want to work on push notifications or use a development build with push notifications, please go the server side code [HERE](https://github.com/home-assistant/mobile-apps-fcm-push) and deploy it to your firebase project. Once you have your androidV1 URL to the deployed service, set it in to your `${GRADLE_USER_HOME}/gradle.properties` file, e.g.:
+```properties
+homeAssistantAndroidPushUrl=https://mydomain.cloudfunctions.net/androidV1
```
Comment (pdecat) [inline README.md:48]: I prefixed the properties with `homeAssistantAndroid` to make it clear which project uses those when overridden in the user's `~/.gradle/gradle.properties`.

**Comment by runningcode** [comment]
Looks great!

Patrick Decat <notifications@github.com> schrieb am So., 11. Okt. 2020,
18:09:

**Review by JBassett — APPROVED** [review]

**Comment by jesserockz** [comment]
A bit late now that its merged, but wouldn't it be nicer to have 1 variable that just overrides the domain name/host part of the url?

**Comment by dshokouhi** [comment]
@jesserockz the firebase URLs have a slightly different URL when you are doing dev so not sure that would work.  Its ok to keep them separate and modify them separately.

**Comment by jesserockz** [comment]
Ah I see what you mean. 
Yeah that happens if you don't deploy the static site which points the correct urls to the functions
- `/api/sendPush/android/v1` -> `androidV1`
- `/api/checkRateLimits` -> `checkRateLimits`

**Pull request merged**